tests/unit/amo/components/TestPage.js

```diff
@@ -240,6 +255,466 @@ describe(__filename, () => {
+        );
+        expect(screen.getByClassName('GetFirefoxBanner')).toHaveClass(
+          'Notice-dismissible',
+        );
+      });
+
+      it('has the expected text', () => {
+        render(props);
+
+        expect(
+          screen.getByTextAcrossTags(
+            `To use these add-ons, you'll need to download Firefox`,
+          ),
+        ).toBeInTheDocument();
+      });
+
+      it('sets the href on the button with the expected utm params', () => {
+        render(props);
+
+        const expectedHref = [
+          `${DOWNLOAD_FIREFOX_BASE_URL}?s=direct`,
+          `utm_campaign=${DOWNLOAD_FIREFOX_UTM_CAMPAIGN}`,
+          `utm_content=${GET_FIREFOX_BANNER_UTM_CONTENT}`,
+          `utm_medium=referral&utm_source=addons.mozilla.org`,
+        ].join('&');
+
+        expect(
+          screen.getByRole('link', { name: 'download Firefox' }),
+        ).toHaveAttribute('href', expectedHref);
+      });
+
+      it('sends a tracking event when the button is clicked', () => {
+        render(props);
+
+        userEvent.click(screen.getByRole('link', { name: 'download Firefox' }));
+
+        expect(tracking.sendEvent).toHaveBeenCalledTimes(1);
+        expect(tracking.sendEvent).toHaveBeenCalledWith({
+          action: GET_FIREFOX_BANNER_CLICK_ACTION,
+          category: GET_FIREFOX_BUTTON_CLICK_CATEGORY,
+        });
+      });
+
+      it('sends a tracking event when the banner is dismissed', () => {
+        render(props);
+
+        userEvent.click(
+          screen.getByRole('button', { name: 'Dismiss this notice' }),
+        );
+
+        expect(tracking.sendEvent).toHaveBeenCalledTimes(1);
+        expect(tracking.sendEvent).toHaveBeenCalledWith({
+          action: GET_FIREFOX_BANNER_DISMISS_ACTION,
+          category: GET_FIREFOX_BANNER_DISMISS_CATEGORY,
+        });
+      });
+    });
+
+    describe('On firefox', () => {
+      it('does not render a download banner if the browser is Firefox Desktop', () => {
+        _dispatchClientMetadata({ userAgent: userAgents.firefox[0] });
+        render(props);
+
+        expect(screen.queryByText('download Firefox')).not.toBeInTheDocument();
+      });
+
+      it('does not render a download banner if the browser is Firefox for Android', () => {
+        _dispatchClientMetadata({ userAgent: userAgents.firefoxAndroid[0] });
+        render(props);
+
+        expect(screen.queryByText('download Firefox')).not.toBeInTheDocument();
+      });
+
+      it('does not render a download banner if the browser is Firefox for iOS', () => {
+        _dispatchClientMetadata({ userAgent: userAgents.firefoxIOS[0] });
+        render(props);
+
+        expect(screen.queryByText('download Firefox')).not.toBeInTheDocument();
+      });
+    });
+  });
+
+  describe('Tests for Header', () => {
+    it('renders an <h1> when isHomepage is true', () => {
+      render({ isHomePage: true });
+
+      expect(
+        screen.getByRole('heading', { name: 'Firefox Browser Add-ons' }),
```

Comment:
afaik `heading` is the role for h1 to h6 so we need to test `aria-level` too.

Reply:
Well, we know that there is only going to be one heading with "Firefox Browser Add-ons" in the page, so I'm not sure we need to assert that it's an `h1`. The test name suggests that's what we are testing, but really we only care that a heading exists with that text. I will update the test description. Ultimately we shouldn't care if it is an `h1` or `h2`, etc.